### PR TITLE
JBIDE-27680: Login/bootstrap Developer Sandbox

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by jeffmaury** [comment]
testPR

**Comment by jeffmaury** [comment]
testPR

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/sandbox/SandboxLoginPage.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    try {
+      if (model.getClusterToken() != null) {
+        status.setValue(ValidationStatus.ok());
+      } else {
+        browser.setUrl(KubernetesClusterHelper.getTokenRequest(model.getClusterURL()));
+        status.setValue(ValidationStatus.cancel("Please complete login"));
+      }
+    } catch (IOException e) {
+      // TODO Auto-generated catch block
+      e.printStackTrace();
```
Comment (sbouchet) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/sandbox/SandboxLoginPage.java:85]: please use proper error logging

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/sandbox/SandboxWorkflowPage.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    return Status.OK_STATUS;
+  }
+  
+  private void launchJob() {
+    try {
+      Job job = Job.create("Retrieving Red Hat Developer Sandbox state", this::retrieveState);
+      WizardUtils.runInWizard(job, getContainer());
+      updateGroups();
+    } catch (InvocationTargetException e) {
+    } catch (InterruptedException e) {
```
Comment (sbouchet) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/sandbox/SandboxWorkflowPage.java:159]: please use multi-catch here.

**Review by sbouchet — APPROVED** [review]
LGTM. can log in the devsandbox using wizard from login page.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/sandbox/SandboxWorkflowPage.java (hunk 2 above)
Comment (jeffmaury) [reply plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/applicationexplorer/sandbox/SandboxWorkflowPage.java:159]: Can't as the logic is different

**Review by sbouchet — APPROVED** [review]
LGTM.

**Pull request merged**